src/pytest_bdd/scenario.py

```diff
@@ -236,34 +237,34 @@ def scenario_wrapper(request: FixtureRequest, _pytest_bdd_example: dict[str, str
 
         scenario_wrapper.__doc__ = f"{feature_name}: {scenario_name}"
         scenario_wrapper.__scenario__ = templated_scenario
-        return cast(Callable, scenario_wrapper)
+        return cast(Callable[P, T], scenario_wrapper)
 
     return decorator
 
 
 def collect_example_parametrizations(
     templated_scenario: ScenarioTemplate,
 ) -> list[ParameterSet] | None:
-    # We need to evaluate these iterators and store them as lists, otherwise
-    # we won't be able to do the cartesian product later (the second iterator will be consumed)
-    contexts = list(templated_scenario.examples.as_contexts())
-    if not contexts:
+    if contexts := list(templated_scenario.examples.as_contexts()):
+        return [pytest.param(context, id="-".join(context.values())) for context in contexts]
+    else:
         return None
```

Comment:
I don't recall having changed this, and I think this is an auto-fix by Ruff. Let me know if you want me to revert this change.

Reply:
this is something Sourcery did, I'll first merge your PR once comments are addressed, then review this one with the correct diff